core/src/main/java/com/datastrato/gravitino/listener/event/Event.java

```diff
@@ -0,0 +1,55 @@
+/*
+ *  Copyright 2024 Datastrato Pvt Ltd.
+ *  This software is licensed under the Apache License version 2.
+ */
+
+package com.datastrato.gravitino.listener.event;
+
+import com.datastrato.gravitino.NameIdentifier;
+import com.datastrato.gravitino.annotation.DeveloperApi;
+
+/**
+ * The abstract base class for all events. It encapsulates common information such as the user who
+ * generated the event and the identifier for the resource associated with the event. Subclasses
+ * should provide specific details related to their individual event types.
+ */
+@DeveloperApi
+public abstract class Event {
+  private String user;
+  private NameIdentifier identifier;
+
+  /**
+   * Constructs an Event instance with the specified user and resource identifier details.
+   *
+   * @param user The user associated with this event. It provides context about who triggered the
+   *     event.
+   * @param identifier The resource identifier associated with this event. This may refer to various
+   *     types of resources such as a metalake, catalog, schema, or table, etc.
```

Comment:
user could distinguish the type of resource by event type

Reply:
I suggest removing the `SupportsAsync` interface and moving its methods to `EventListenerPlugin.java` and give the default implementation, so developers only need to focus on `EventListenerPlugin`.

Reply:
> I suggest removing the `SupportsAsync` interface and moving its methods to `EventListenerPlugin.java` and give the default implementation, so developers only need to focus on `EventListenerPlugin`. WDYT @jerryshao

The comment is unrelated here. About removing `SupportsAsync`, do you have any better design on how to do this? Yes, the current `SupportsAsync` is not so intuitive to users, but I don't have a strong opinion on it.



> I am a little confused, how can users distinguish what type of resource it is from `NameIdentifier`?

Regarding this, I think this can be distinguished by event class typel

Reply:
Apologies for commenting in the wrong place. Here is my recommendation: https://github.com/datastrato/gravitino/pull/2787#discussion_r1560428279